src/wallet/wallet.cpp

```diff
@@ -1178,7 +1178,7 @@ bool CWallet::SetHDMasterKey(const CKey& key)
     // key metadata is not required
     CPubKey pubkey = key.GetPubKey();
     if (!AddKeyPubKey(key, pubkey))
-        throw std::runtime_error("CWallet::GenerateNewKey(): AddKey failed");
+        throw std::runtime_error("CWallet::SetHDMasterKey(): AddKey failed");
```

Comment:
Nit: if you do want to keep the function name here, please use `strprintf("%s: AddKey failed", __func__)` instead of hardcoding it.